File: apps/mobile/types/pet.ts
export interface Pet {
  id: string;
  owner_id: string;
  name: string;
  species: string;
  breed: string;
  gender: string;
  birth_date?: string;
  weight?: number;
  avatar_url: string;
  microchip?: string;
  is_neutered: boolean;
  allergies: string[];
  notes: string;
  health_score?: number;
  created_at: string;
  updated_at: string;
}

export interface CreatePetPayload {
  name: string;
  species: string;
  breed?: string;
  gender?: string;
  weight?: number;
  avatar_url?: string;
  is_neutered?: boolean;
  allergies?: string[];
  notes?: string;
}

File: server/internal/model/pet.go
package model

import (
	"time"

	"github.com/google/uuid"
	"gorm.io/datatypes"
	"gorm.io/gorm"
)

type Pet struct {
	ID          uuid.UUID `gorm:"type:uuid;primaryKey;default:gen_random_uuid()"`
	OwnerID     uuid.UUID `gorm:"type:uuid;index;not null"`
	Name        string    `gorm:"size:50;not null"`
	Species     string    `gorm:"size:20;not null"`
	Breed       string    `gorm:"size:50"`
	Gender      string    `gorm:"size:10"`
	BirthDate   *time.Time
	Weight      *float64
	AvatarURL   string         `gorm:"size:500"`
	Microchip   *string        `gorm:"size:50"`
	IsNeutered  bool           `gorm:"default:false"`
	Allergies   datatypes.JSON `gorm:"type:jsonb"`
	Notes       string         `gorm:"type:text"`
	HealthScore *int
	CreatedAt   time.Time
	UpdatedAt   time.Time
	DeletedAt   gorm.DeletedAt `gorm:"index"`
	Owner       User           `gorm:"foreignKey:OwnerID"`
}

func (p *Pet) BeforeCreate(_ *gorm.DB) error {
	if p.ID == uuid.Nil {
		p.ID = uuid.New()
	}
	return nil
}
